packages/shared/src/components/sidebar/Sidebar.tsx

```diff
@@ -156,10 +165,30 @@ export default function Sidebar({
     optOutWeeklyGoal,
   } = useContext(SettingsContext);
   const [showSettings, setShowSettings] = useState(false);
+  const [showIntroModal, setShowIntroModal] = useState(false);
+
   const shouldShowDnD = !!process.env.TARGET_BROWSER;
   const { flags } = useContext(FeaturesContext);
   const popularFeedCopy = getFeatureValue(Features.PopularFeedCopy, flags);
   const feedFilterModal = getFeatureValue(Features.FeedFilterModal, flags);
+  const feedFilterModalOnboarding = getFeatureValue(
+    Features.FeedFilterModalOnboarding,
+    flags,
+  );
+  const [isFirstSession, setIsFirstSession, isSessionLoaded] =
+    usePersistentContext(FIRST_TIME_SESSION, true);
```

Comment:
Two comments:
* I think this should be in some context rather than in the sidebar. Because other components might need it as well.
* This is not so bulletproof because existing users will also see this dialog and it's not the expected behavior. We want only new users to see it. We have to think about another way to achieve it. It's a global application challenge I think. How to identify a new user? I think you need some Gateway work here as well.

Reply:
I'll raise a sub ticket for this to identify first time users.
Think it's a bit out of scope for Aya to create this.

Reply:
However raising another question, should it not simply show for all users who DON'T have feed filters yet?
Rather than first time visitors?

Reply:
Some of the variants here don't make sense for existing users as they will make the app not usable for someone who didn't filter. It might be too strict in my opinion but I'd love to hear what you think?